src/Compression/Compression.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+using ObjCRuntime;
+
+namespace Compression {
+
+
+	public partial class CompressionStream : IDisposable {
+
+		[StructLayout(LayoutKind.Sequential)]
+		struct CompressionStreamStruct {
+			public IntPtr Destination;
+			public int DestinationLength;
+			public IntPtr Source;
+			public int SourceLength;
+			public IntPtr State;
+		}
+
+		[DllImport (Constants.libcompression)]
+		static extern int /* CompressionStatus */ CompressionStreamInit (ref CompressionStreamStruct stream, int /* StreamOperation */ operation, int /* CompressionAlgorithm */ algorithm);
```

Comment:
How does this work? The native method is called `compression_stream_init`, not `CompressionStreamInit` (and the same for the other P/Invokes).

Reply:
This is from and old hash. I guess that github did not pick that up, should be ok now :/